alembic/versions/2023_07_18_1727-2e42e5319395_md5sum_fileondisk.py

```diff
@@ -0,0 +1,42 @@
+"""md5sum_fileondisk
+
+Revision ID: 2e42e5319395
+Revises: b90b1e3ec58c
+Create Date: 2023-07-18 17:27:22.177269
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '2e42e5319395'
+down_revision = 'b90b1e3ec58c'
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('cutouts', sa.Column('md5sum', sa.UUID(), nullable=True))
+    op.add_column('cutouts', sa.Column('md5sum_extensions', sa.ARRAY(sa.UUID()), nullable=True))
+    op.add_column('exposures', sa.Column('md5sum', sa.UUID(), nullable=True))
+    op.add_column('exposures', sa.Column('md5sum_extensions', sa.ARRAY(sa.UUID()), nullable=True))
+    op.add_column('images', sa.Column('md5sum', sa.UUID(), nullable=True))
+    op.add_column('images', sa.Column('md5sum_extensions', sa.ARRAY(sa.UUID()), nullable=True))
+    op.add_column('source_lists', sa.Column('md5sum', sa.UUID(), nullable=True))
+    op.add_column('source_lists', sa.Column('md5sum_extensions', sa.ARRAY(sa.UUID()), nullable=True))
```

Comment:
We should make the design decision.  I was operating uner the idea that we might in code create some database entries before saving to the archive, and then the existing of a non-NULL md5sum would be the flag that it was on the archive.

Unless we think of a reason why we would need to do that, yeah, it might be better to demand that something is saved to the archive before we save it to the database.  We'll have to make sure that there isn't any code written already that would violate that assumption.

Reply:
...see comment on exposure.py save() below, which is relevant to this.